<?xml version="1.0"?>
<launch>

<arg name="visualize" default="true" />

<node pkg="graph_localization" type="graph_localization_node" name="graph_localization_node" output="screen">

<param name="localisation_type" value="mcl_ndt" />
<param name="check_consistency" value="true" />

<param name="gt_topic" value="/vmc_navserver/state" />
<param name="pose_init_x" value="10.73" />
<param name="pose_init_y" value="2.185" />
<param name="pose_init_t" value="0.02" />

<param name="init_pose_ros" value="true" />
<param name="init_pose_gt" value="false" />

<param name="localisation_2d" value="true" />
<param name="scanner_2d" value="false" />
<param name="use_odometry" value="true" />

<param name="odometry_topic" value="/vmc_navserver/odom" />
<param name="velodyne_topic" value="/velodyne_packets" />
<param name="laser_topic" value="/laserscan" />

<param name="z_filter_height" value="-101" />
<param name="n_particles" value="251" />
<param name="SIR_max_iters_wo_resampling" value="31" />
<param name="forceSIR" value="true" />

<!--<param name="laser_variance_z" value="0.02" / -->
<param name="max_range" value="130." />
<param name="min_range" value="1.0" />

<!-- Specific sensor offset parameters for the example file -->
<param name="sensor_pose_x" value="1.17920618535632" />
<param name="sensor_pose_y" value="-0.285884882359476" />
<param name="sensor_pose_z" value="2.0" />
<param name="sensor_pose_t" value="-1.58804135060281" />
<param name="sensor_offset_t" value="0.025" />



<param name="map_path" value="/home/daniel/.ros/maps/offarla-2012_gt=0_submap=0_sizexy=500_Z=12_intrchR=4_compR=0_res=0.4_maxSensd=130_keyF=1_d=0.3_deg=3_alpha=0_dl=0_xyzir=0_mpsu=0_mnpfg=6kmnp0.map" />

<param name="velodyne_config_file" value="$(find ndt_offline)/config/velo32.yaml" />

<param name="plot_marker" value="point" />

<param name="use_keyframe" value="true" />
<param name="min_keyframe_dist" value="0.5" />
<param name="min_keyframe_rot_deg" value="5.0" />
</node>

<group if="$(arg visualize)">
<node pkg="rviz" type="rviz" name="rviz" args="-d $(find graph_map)/launch/3d.rviz" />
</group>


<arg name="path" default="$(find ndt_offline)/bag/arla_bags/velodyne_mapping/" />
<arg name="file_1" default="2012-09-05-09-44-50_0.bagfix.bag" />
<node pkg="rosbag" type="play" name="player" output="screen" args="--clock -r 0.5 -q $(arg path)$(arg file_1)"/>

<!--<node pkg="tf2_ros" type="static_transform_publisher" name="laser_scan_fuser" args="1.17920618535632 -0.3 2.0 0 0 -0.7242871743701824 0.6894984329517051 state_base_link velodyne" />-->

</launch>
<!--
<node pkg="tf2_ros" type="static_transform_publisher" name="laser_scan_fuser" args="0.3 0 0 0 0 -0.72429 0.6895 state_base_link velodyne" / -->
